File: scripts/flush-kv.mjs
#!/usr/bin/env node
/**
 * Flush KV Namespace
 * 
 * Deletes all keys from TASKS_KV namespace.
 * ⚠️  WARNING: This is destructive! Make sure you have a backup first.
 * 
 * Usage:
 *   node scripts/flush-kv.mjs
 * 
 * With confirmation bypass (for CI):
 *   FLUSH_CONFIRM=yes node scripts/flush-kv.mjs
 */

import { execSync } from 'child_process';
import readline from 'readline';

const KV_NAMESPACE = 'TASKS_KV';

console.log('⚠️  KV NAMESPACE FLUSH\n');
console.log('This will DELETE ALL KEYS from the TASKS_KV namespace.');
console.log('Make sure you have created a backup first!\n');

// Check for auto-confirm (for CI)
const autoConfirm = process.env.FLUSH_CONFIRM === 'yes';

async function confirm() {
  if (autoConfirm) {
    console.log('Auto-confirmed via FLUSH_CONFIRM environment variable\n');
    return true;
  }
  
  const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
  });
  
  return new Promise((resolve) => {
    rl.question('Type "FLUSH" to confirm deletion: ', (answer) => {
      rl.close();
      resolve(answer === 'FLUSH');
    });
  });
}

async function main() {
  const confirmed = await confirm();
  
  if (!confirmed) {
    console.log('❌ Flush cancelled');
    process.exit(0);
  }
  
  console.log('\n🔍 Listing keys to delete...');
  
  try {
    // List all keys
    const listCommand = `npx wrangler kv key list --binding=${KV_NAMESPACE}`;
    const keysJson = execSync(listCommand, { 
      cwd: 'workers/task-api',
      encoding: 'utf-8' 
    });
    
    const keys = JSON.parse(keysJson);
    console.log(`✅ Found ${keys.length} keys to delete\n`);
    
    if (keys.length === 0) {
      console.log('✨ KV namespace is already empty');
      process.exit(0);
    }
    
    // Delete each key
    console.log('🗑️  Deleting keys...');
    let count = 0;
    let failed = 0;
    
    for (const { name: key } of keys) {
      try {
        const deleteCommand = `npx wrangler kv key delete "${key}" --binding=${KV_NAMESPACE}`;
        execSync(deleteCommand, { 
          cwd: 'workers/task-api',
          stdio: 'pipe' // Suppress output
        });
        
        count++;
        process.stdout.write(`\r   Progress: ${count}/${keys.length} keys deleted`);
      } catch (error) {
        failed++;
        console.error(`\n❌ Failed to delete key: ${key}`, error.message);
      }
    }
    
    console.log('\n');
    console.log(`✅ Flush completed!`);
    console.log(`   Successfully deleted: ${count} keys`);
    if (failed > 0) {
      console.log(`   Failed: ${failed} keys`);
    }
    
    console.log('\n✨ KV namespace flushed!');
    
  } catch (error) {
    console.error('\n❌ Flush failed:', error.message);
    process.exit(1);
  }
}

main();

File: scripts/flush-kv.py
#!/usr/bin/env python3
"""KV Flush Script"""
import requests, os, sys
from pathlib import Path

def load_config():
    env_path = Path(__file__).parent.parent / '.env'
    config = {'CLOUDFLARE_API_TOKEN': os.environ.get('CLOUDFLARE_API_TOKEN'),
              'CLOUDFLARE_ACCOUNT_ID': os.environ.get('CLOUDFLARE_ACCOUNT_ID'),
              'CLOUDFLARE_NAMESPACE_ID': os.environ.get('CLOUDFLARE_NAMESPACE_ID')}
    if not all(config.values()) and env_path.exists():
        with open(env_path) as f:
            for line in f:
                if '=' in line and not line.strip().startswith('#'):
                    k, v = line.split('=', 1)
                    k, v = k.strip(), v.strip().strip('"').strip("'")
                    if k in config and not config[k]: config[k] = v
    return config

def main():
    print("WARNING: This will delete ALL keys from KV!")
    cfg = load_config()
    if not all([cfg['CLOUDFLARE_API_TOKEN'], cfg['CLOUDFLARE_ACCOUNT_ID'], cfg['CLOUDFLARE_NAMESPACE_ID']]):
        print("Error: Missing credentials"); return 1
    
    url = f"https://api.cloudflare.com/client/v4/accounts/{cfg['CLOUDFLARE_ACCOUNT_ID']}/storage/kv/namespaces/{cfg['CLOUDFLARE_NAMESPACE_ID']}/keys"
    headers = {'Authorization': f'Bearer {cfg['CLOUDFLARE_API_TOKEN']}'}
    
    all_keys, cursor = [], None
    while True:
        params = {'limit': 1000}
        if cursor: params['cursor'] = cursor
        r = requests.get(url, headers=headers, params=params)
        if r.status_code != 200: print(f"Error: {r.status_code}"); return 1
        data = r.json()
        all_keys.extend([k['name'] for k in data.get('result', [])])
        cursor = data.get('result_info', {}).get('cursor')
        if not cursor: break
    
    print(f"Found {len(all_keys)} keys to delete")
    if not all_keys: print("Nothing to delete"); return 0
    
    if os.environ.get('FLUSH_CONFIRM') != 'yes':
        confirm = input('Type "FLUSH" to confirm deletion: ')
        if confirm != 'FLUSH': print("Cancelled"); return 1
    
    for i, key in enumerate(all_keys, 1):
        delete_url = f"https://api.cloudflare.com/client/v4/accounts/{cfg['CLOUDFLARE_ACCOUNT_ID']}/storage/kv/namespaces/{cfg['CLOUDFLARE_NAMESPACE_ID']}/values/{key}"
        requests.delete(delete_url, headers=headers)
        print(f"Progress: {i}/{len(all_keys)}", end='\r')
    
    print(f"\nFlush complete: {len(all_keys)} keys deleted")
    return 0

if __name__ == '__main__':
    sys.exit(main())
